keras_nlp/layers/modeling/lora_dense_test.py

```diff
@@ -0,0 +1,117 @@
+from keras_nlp.tests.test_case import TestCase
+
+
+@pytest.mark.multi_backend_only
+class LoraDenseTest(TestCase):
+    def test_layer_behaviors(self):
+        self.run_layer_test(
+            layer_cls=LoraDense,
+            init_kwargs={
+                "inner_dense": keras.layers.Dense(16),
+                "rank": 2,
+                "alpha": 16,
+                "lora_a_initializer": "HeNormal",
+            },
+            input_data=ops.random.uniform(shape=(2, 4, 8)),
+            expected_output_shape=(2, 4, 16),
+            expected_num_trainable_weights=2,
+            expected_num_non_trainable_weights=2,
+            expected_num_non_trainable_variables=2,
+            run_mixed_precision_check=False,
+        )
+
+    def test_layer_behaviors_einsum(self):
+        self.run_layer_test(
+            layer_cls=LoraDense,
+            init_kwargs={
+                "inner_dense": keras.layers.EinsumDense(
+                    "abc,cde->abde",
+                    output_shape=(None, 2, 16),
+                ),
+                "lora_a_initializer": "HeNormal",
+            },
+            input_data=ops.random.uniform(shape=(2, 4, 8)),
+            expected_output_shape=(2, 4, 2, 16),
+            expected_num_trainable_weights=2,
+            expected_num_non_trainable_weights=1,
+            expected_num_non_trainable_variables=1,
+            run_mixed_precision_check=False,
+        )
+
+    def test_merge_dense(self):
+        inner_dense = keras.layers.Dense(16)
+        layer = LoraDense(inner_dense, rank=4)
+        layer.build((2, 16))
+        layer.lora_a.assign(ops.random.uniform(shape=(16, 4)))
+        layer.lora_b.assign(ops.random.uniform(shape=(4, 16)))
+
+        input_data = ops.random.uniform((2, 16))
+        lora_output = layer(input_data)
+        dense_output = inner_dense(input_data)
+        self.assertNotAllClose(lora_output, dense_output)
+
+        layer.merge_weights()
+        merged_lora_output = layer(input_data)
+        dense_output = inner_dense(input_data)
+        self.assertAllClose(lora_output, merged_lora_output)
+        self.assertAllClose(lora_output, dense_output)
+
+    def test_merge_einsum(self):
+        inner_dense = keras.layers.EinsumDense(
+            "abc,cde->abde",
+            output_shape=(None, 2, 16),
+        )
+        layer = LoraDense(inner_dense, rank=4)
+        layer.build((2, 4, 16))
+        layer.lora_a.assign(ops.random.uniform(shape=(16, 4)))
+        layer.lora_b.assign(ops.random.uniform(shape=(4, 2, 16)))
+
+        input_data = ops.random.uniform((2, 4, 16))
+        lora_output = layer(input_data)
+        dense_output = inner_dense(input_data)
+        self.assertNotAllClose(lora_output, dense_output)
+
+        layer.merge_weights()
+        merged_lora_output = layer(input_data)
+        dense_output = inner_dense(input_data)
+        self.assertAllClose(lora_output, merged_lora_output)
+        self.assertAllClose(lora_output, dense_output)
```

Comment:
It might be worthwhile to include a test case about the trainability of all the weights included in the layer, especially since these "freeze" params exist and they're acted upon in `build`